### PR TITLE
document how to modify OE submodule

### PR DESCRIPTION
**Pull request by thomasten**

### CODE REVIEW

File: 3rdparty/openenclave/HACK.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Enabling OEDEV will leave the `openenclave` subfolder as is (instead of copying it to the build folder and applying the patch), so modifications can immediately built and tested.
+
+Open a second instance of your IDE at `3rdparty/openenclave/openenlave` and apply the patch:
+```sh
+patch -p1 < ../ert.patch
+git commit -a -m ert
+```
+Now you can make changes to both ERT and OE. You only need to build ERT which automatically builds the modified OE.
+
+Once you are done, create a new patch file (within the IDE opened at OE):
```
Comment (Nirusu) [inline 3rdparty/openenclave/HACK.md:19]:
```suggestion
Once you are done, create a new patch file (within `3rdparty/openenclave/openenclave`):
```
Think it makes sense here to specify the exact directory, as if we are somewhere deeper in OE, the patch will be written to the wrong directory.

**Review by Nirusu — COMMENTED** [review]
`HACK.md`, mhh... Not really sure I like the file name. It sound luring to look inside from the file viewer, but I would not exactly expect instructions to build the patch files in here.

Maybe `DEVELOPMENT.md`, or `PATCH.md`?

**Comment by thomasten** [comment]
> 
> 
> `HACK.md`, mhh... Not really sure I like the file name. It sound luring to look inside from the file viewer, but I would not exactly expect instructions to build the patch files in here.
> 
> Maybe `DEVELOPMENT.md`, or `PATCH.md`?

This was inspired by https://github.com/aquynh/capstone/blob/master/HACK.TXT. Other projects use HACK or HACKING for their development instructions, too, e.g., https://github.com/snapcore/snapd/blob/master/HACKING.md or https://github.com/ghc/ghc/blob/master/HACKING.md.

So what do you think? HACK, HACKING, DEVELOPMENT, DEVELOP, DEVELOPING? (I don't like PATCH)

**Comment by Nirusu** [comment]
Didn't really notice it was this common, but actually it is. You never stop learning...

Would go with HACKING.md, that seems to be the most common variant.

**Review by Nirusu — APPROVED** [review]

**Pull request merged**